scripts/lint/test-encoder-null-safety.sh

```
@@ -0,0 +1,113 @@
+  fi
+
+  if [[ "$pass_label" == "PASS" ]]; then
+    echo "PASS: $label (exit=$rc)"
+  else
+    echo "FAIL: $label ($mode)"
+    echo "  expected exit: $expected_exit  got: $rc"
+    if [[ -n "$expected_stdout_match" ]]; then
+      echo "  expected output to contain: $expected_stdout_match"
+    fi
+    echo "  --- stdout/stderr ---"
+    echo "$out" | sed 's/^/    /'
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)